src/main.js

```diff
@@ -4,18 +4,23 @@
  */
 
 import React from 'react';
-import { AppRegistry, YellowBox } from 'react-native';
+import { AppRegistry, YellowBox, BackHandler } from 'react-native';
 import { Provider } from 'react-redux';
 import { NavigationContainer } from '@react-navigation/native';
 import { ErrorHandler } from 'redux-persist-error-handler';
 
 import { name as appName } from '../app.json';
 import { store, persistedStore } from './Store';
-import { rootNavigatorRef } from './navigation';
+import { backHandler, rootNavigatorRef } from './navigation';
 
 import MSupplyMobileApp from './mSupplyMobileApp';
 
-function App() {
+const App = () => {
+  React.useEffect(() => {
+    BackHandler.addEventListener('hardwareBackPress', backHandler(store));
+    return () => BackHandler.removeEventListener('hardwareBackPress', backHandler(store));
+  }, []);
```

Comment:
This only occurs when this component first mounts and then the return function is run when it's unmounted. Injecting the store into the back handler function to avoid a cycle, mainly